azurerm/resource_arm_api_management.go

```diff
@@ -711,7 +712,7 @@ func resourceArmApiManagementServiceDelete(d *schema.ResourceData, meta interfac
 	log.Printf("[DEBUG] Deleting API Management Service %q (Resource Grouo %q)", name, resourceGroup)
 	resp, err := client.Delete(ctx, resourceGroup, name)
 	if err != nil {
-		if !utils.ResponseWasNotFound(resp) {
+		if !(resp.Response() != nil && (*resp.Response()).StatusCode == http.StatusNotFound) { // TODO: VERIFY!!!
```

Comment:
The new API returns a Future type. The helper did not have support for that. 
I think this codes does the same logical check, i.e. check that the response is _not_ 404.

Reply:
I don't get it.
The new API already excludes 404
https://github.com/Azure/azure-sdk-for-go/blob/1a2940b1f9d783e439ac76230b3ecc983a8cc0b3/services/apimanagement/mgmt/2019-01-01/apimanagement/service.go#L127-L136

Reply:
You referenced the code for the _create_ function, this is the _delete_ function.

https://github.com/Azure/azure-sdk-for-go/blob/1a2940b1f9d783e439ac76230b3ecc983a8cc0b3/services/apimanagement/mgmt/2019-01-01/apimanagement/service.go#L419-L451

It is different than in the 2018 API
https://github.com/Azure/azure-sdk-for-go/blob/1a2940b1f9d783e439ac76230b3ecc983a8cc0b3/services/apimanagement/mgmt/2018-01-01/apimanagement/service.go#L419-L456